numpy/core/tests/test_datetime.py

```diff
@@ -1081,6 +1085,26 @@ def test_datetime_compare(self):
         assert_equal(np.greater(a, b), [0, 1, 0, 1, 0])
         assert_equal(np.greater_equal(a, b), [1, 1, 0, 1, 0])
 
+    def test_datetime_compare_nat(self):
+        dt_nat = np.datetime64('NaT', 'D')
+        dt_other = np.datetime64('2000-01-01')
+        td_nat = np.timedelta64('NaT', 'h')
+        td_other = np.timedelta64(1, 'h')
+        for op in [np.equal, np.less, np.less_equal,
+                   np.greater, np.greater_equal]:
+            assert not op(dt_nat, dt_nat)
```

Comment:
will do... I would be surprised if anyone is running our test suite with runtime optimized python though :)